machine/quanta/quanta_common_p2020/u-boot/platform-quanta-common_p2020.patch

```diff
@@ -0,0 +1,2235 @@
+index 0000000..4d7edca
+--- /dev/null
++++ b/board/quanta/quanta_common_p2020/law.c
+@@ -0,0 +1,32 @@
++/*
++ * Copyright 2010-2011 Freescale Semiconductor, Inc.
++ *
++ * See file CREDITS for list of people who contributed to this
++ * project.
++ *
++ * This program is free software; you can redistribute it and/or
++ * modify it under the terms of the GNU General Public License as
++ * published by the Free Software Foundation; either version 2 of
++ * the License, or (at your option) any later version.
++ *
++ * This program is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++ * GNU General Public License for more details.
++ *
++ * You should have received a copy of the GNU General Public License
++ * along with this program; if not, write to the Free Software
++ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
++ * MA 02111-1307 USA
++ */
++
++#include <common.h>
++#include <asm/fsl_law.h>
++#include <asm/mmu.h>
++
++struct law_entry law_table[] = {
++	SET_LAW(CONFIG_SYS_FLASH_BASE_PHYS, LAW_SIZE_64M, LAW_TRGT_IF_LBC),
++	SET_LAW(CONFIG_SYS_FLASH_BASE_PHYS1, LAW_SIZE_64M, LAW_TRGT_IF_LBC),
++};
++
++int num_law_entries = ARRAY_SIZE(law_table);
+diff --git a/board/quanta/quanta_common_p2020/mb_detect.c b/board/quanta/quanta_common_p2020/mb_detect.c
+new file mode 100644
+index 0000000..5fa2e7b
+--- /dev/null
++++ b/board/quanta/quanta_common_p2020/mb_detect.c
+@@ -0,0 +1,132 @@
++#include <common.h>
++#include <i2c.h>
++#include <mb_detect.h>
++
++u8 motherboard_id = MAX_NUM_MB;
++u8 motherboard_index = MAX_NUM_MB;
++u8 cpuboard_capability[] = "ly6,ly8,ly5a,ly2a,ly9,ly9a,ly6f,p05,ix1,ix2,p05a";
++SysBoardInfo sysbdinfo;
++SysEEPROMPath eepromPath;
++
++MBDetectPath mbDetectPath[] = {
++	{ 0x72, 2, 0x23, I2C_DEVICE_PCA9555_LENGTH, 1, 2 },
++	{ 0x77, 0, 0x23, I2C_DEVICE_PCA9555_LENGTH, 1, 2 },
++	{ 0x77, 7, 0x23, I2C_DEVICE_PCA9555_LENGTH, 1, 2 },
++	{ 0x77, 2, 0x23, I2C_DEVICE_PCA9555_LENGTH, 1, 2 },
++};
++
++SysBoardInfo sysBoardInfo[MAX_NUM_MB] = {
++	{ 0x00, "ly6_p2020", "LY6\0", "LY6-P2020\0", 0x0000f601 },
++	{ 0x01, "ly8_p2020", "LY8\0", "LY8-P2020\0", 0x0000f801 },
++	{ 0x09, "ly5a_p2020", "LY5A\0", "LY5A-P2020\0", 0x0000f5a1 },
++	{ 0x08, "ly2a_p2020", "LY2A\0", "LY2A-P2020\0", 0x0000f2a1 },
++	{ 0x02, "ly9_p2020",  "LY9\0",  "LY9-P2020\0",  0x0000f901 },
++	{ 0x0e, "ly9a_p2020",  "LY9A\0",  "LY9A-P2020\0",  0x0000f9a1 },
++	{ 0x05, "ly6f_p2020",  "LY6F\0",  "LY6F-P2020\0",  0x0000f6f1 },
++	{ 0x11, "p05_p2020",  "P05\0",  "P05-P2020\0",  0x0000d501 },
++	{ 0x0d, "ix1_p2020",  "IX1\0",  "IX1-P2020\0",   0x0000c101 },
++	{ 0x0f, "ix2_p2020",  "IX2\0",  "IX2-P2020\0",  0x0000c201 },
++	{ 0x12, "p05a_p2020",  "P05A\0",  "P05A-P2020\0",  0x0000d5a1 },
++};
++
++SysEEPROMPath sysEEPROMPath[] = {
++	{ 0x72, 2, 0x54},
++	{ 0x72, 3, 0x54},
++	{ 0x77, 2, 0x54},
++};
++
++int mb_index_get(void) {
```

Comment:
Code quality problems -- too much reliance on global variables

This function returns an 'int', yet below when called below the result is ignored.  Instead the proper functioning relies on the setting of global variables `sysbdinfo` and `eepromPath`.

This function also uses a global variable `motherboard_id` as an input variable.  Why not just pass that value into the function on the stack?

This function, its callers and related data structures should be redesigned to get rid of the global variables and confusing code paths.

One example would be to include the sysEEPROMPath[] info in the sysBoardInfo[] structure.